crates/uv-distribution-types/src/index_url.rs

```diff
@@ -258,21 +323,10 @@ impl<'de> serde::de::Deserialize<'de> for IndexUrl {
 }
 
 impl From<VerbatimUrl> for IndexUrl {
-    fn from(mut url: VerbatimUrl) -> Self {
-        if url.scheme() == "file" {
-            Self::Path(Arc::new(url))
-        } else {
-            // Remove trailing slashes for consistency. They'll be re-added if necessary when
-            // querying the Simple API.
-            if let Ok(mut path_segments) = url.raw_mut().path_segments_mut() {
-                path_segments.pop_if_empty();
-            }
-            if *url.raw() == *PYPI_URL {
-                Self::Pypi(Arc::new(url))
-            } else {
-                Self::Url(Arc::new(url))
-            }
-        }
+    fn from(url: VerbatimUrl) -> Self {
+        // Remove trailing slashes for consistency. They'll be re-added if necessary when
+        // querying the Simple API.
+        Self::from_verbatim_url_with_trailing_slash_policy(url, TrailingSlashPolicy::Remove)
     }
```

Comment:
I think we should just get rid of this, since it seems like callers _must_ be cognizant of whether they want to preserve or remove the trailing slash.